runtime/integration-tests/src/sibling.rs

```diff
@@ -0,0 +1,613 @@
+};
+use pallet_xcm::XcmPassthrough;
+use polkadot_parachain::primitives::Sibling;
+use polkadot_runtime_common::MAXIMUM_BLOCK_WEIGHT;
+use runtime_common::parachains::kusama::asset_hub;
+use scale_info::TypeInfo;
+use serde::{Deserialize, Serialize};
+use sp_core::H256;
+use sp_debug_derive::RuntimeDebug;
+use sp_runtime::{
+	testing::Header,
+	traits::{BlakeTwo256, Convert, IdentityLookup, Zero},
+	AccountId32,
+};
+use xcm::v3::prelude::*;
+use xcm_emulator::{
+	cumulus_pallet_parachain_system::{self, RelayNumberStrictlyIncreases},
+	Weight,
+};
+use xcm_executor::{
+	traits::{JustTry, ShouldExecute, WeightTrader},
+	Assets, XcmExecutor,
+};
+
+use xcm::latest::Weight as XCMWeight;
+use xcm_builder::{
+	AccountId32Aliases, AllowUnpaidExecutionFrom, ConvertedConcreteId, EnsureXcmOrigin,
+	FixedWeightBounds, FungiblesAdapter, NoChecking, ParentIsPreset, RelayChainAsNative,
+	SiblingParachainAsNative, SiblingParachainConvertsVia, SignedAccountId32AsNative,
+	SignedToAccountId32, SovereignSignedViaLocation,
+};
+
+use crate::{AMPLITUDE_ID, KUSAMA_ASSETHUB_ID, PENDULUM_ID, POLKADOT_ASSETHUB_ID};
+
+const XCM_ASSET_RELAY_DOT: u8 = 0;
+const XCM_ASSET_ASSETHUB_USDT: u8 = 1;
+
+pub type AccountId = AccountId32;
+
+parameter_types! {
+	pub const RelayLocation: MultiLocation = MultiLocation::parent();
+	pub const RelayNetwork: NetworkId = NetworkId::Polkadot;
+	pub RelayChainOrigin: RuntimeOrigin = cumulus_pallet_xcm::Origin::Relay.into();
+	pub CheckingAccount: AccountId = PolkadotXcm::check_account();
+	pub UniversalLocation: InteriorMultiLocation =
+		X2(GlobalConsensus(RelayNetwork::get()), Parachain(ParachainInfo::parachain_id().into()));
+
+}
+
+/// Type for specifying how a `MultiLocation` can be converted into an `AccountId`. This is used
+/// when determining ownership of accounts for asset transacting and when attempting to use XCM
+/// `Transact` in order to determine the dispatch Origin.
+pub type LocationToAccountId = (
+	// The parent (Relay-chain) origin converts to the parent `AccountId`.
+	ParentIsPreset<AccountId>,
+	// Sibling parachain origins convert to AccountId via the `ParaId::into`.
+	SiblingParachainConvertsVia<Sibling, AccountId>,
+	// Straight up local `AccountId32` origins just alias directly to `AccountId`.
+	AccountId32Aliases<RelayNetwork, AccountId>,
+);
+
+#[derive(
+	Encode,
+	Decode,
+	Eq,
+	PartialEq,
+	Copy,
+	Clone,
+	RuntimeDebug,
+	PartialOrd,
+	Ord,
+	MaxEncodedLen,
+	TypeInfo,
+)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum CurrencyId {
+	Pendulum,
+	Amplitude,
+	Native,
+	XCM(u8),
+}
+
+// Convert from u32 parachain id to CurrencyId
+// Needed for the test macro so it works regardless of the XCM sender parachain
+impl From<u32> for CurrencyId {
+	fn from(id: u32) -> Self {
+		match id {
+			PENDULUM_ID => CurrencyId::Pendulum,
+			AMPLITUDE_ID => CurrencyId::Amplitude,
+			KUSAMA_ASSETHUB_ID | POLKADOT_ASSETHUB_ID => CurrencyId::XCM(XCM_ASSET_ASSETHUB_USDT),
```

Comment:
ATM both Kusama and Polkadot asset hub have the same ids and this gives a warning of "unreachable" for the second pattern. I left it here in case we change the ids at some point. In that case, the warning will disappear.